libraries/browser-tracker-core/src/tracker/out_queue.ts

```diff
@@ -418,9 +411,13 @@ export function OutQueueManager(
           // When beaconStatus is true, we can't _guarantee_ that it was successful (beacon queues asynchronously)
           // but the browser has taken it out of our hands, so we want to flush the queue assuming it will do its job
           if (beaconStatus === true) {
-            onPostSuccess(numberToSend);
+            removeEventsFromQueue(numberToSend);
+            executeQueue();
+            onRequestSuccess?.(batch);
```

Comment:
I think it'd be good to be consistent about the order of execution of these actions – probably the callback should go before `executeQueue`. Above it's called even before removing events from queue.